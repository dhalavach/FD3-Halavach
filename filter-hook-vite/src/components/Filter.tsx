import { useState, useEffect } from 'react';
import Controls from './Controls';
import List from './List';

export default function Filter(props: { data: string[] }) {
  const { data } = props;

  const [items, setItems] = useState(data);
  const [isSorted, setIsSorted] = useState(false);
  const [searchQuery, setSearchQuery] = useState("");

  const handleSearchQuery = (query: string) => {
    setSearchQuery(query);
  };

  const handleChange = (checked: Boolean) => {
    setIsSorted(Boolean(checked));
  };

  const handleReset = () => {
    setIsSorted(false);
    setSearchQuery('');
  };

  const sortList = (arr: string[]) => {
    const collator = new Intl.Collator('en', {
      numeric: true,
      sensitivity: 'base',
    });
    return arr.sort((a: string, b: string) => {
      return collator.compare(a, b);
    });
  };

  useEffect(() => {
    let arr = [...data];
    if (searchQuery) {
      arr = arr.filter((item) => {
        return item.toLowerCase().includes(searchQuery.toLowerCase());
      });
    }
    if (isSorted) sortList(arr);
    setItems(arr);
  }, [isSorted, searchQuery]);

  return (
    <>
      <Controls
        searchQuery={searchQuery}
        isSorted={isSorted}
        handleSearchQuery={handleSearchQuery}
        handleChange={handleChange}
        handleReset={handleReset}
      />
      <List items={items} />
    </>
  );
}
