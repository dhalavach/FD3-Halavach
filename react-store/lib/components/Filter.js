"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Filter;
function Filter(props) {
  var count = props.count,
    type = props.type,
    handleSortProducts = props.handleSortProducts,
    handleFilterProducts = props.handleFilterProducts,
    sort = props.sort;
  return <div>
      <div className='filter-result'>{count} Product(s)</div>
      <div className='filter-sort'>
        Order{' '}
        <select value={sort} onChange={handleSortProducts}>
          <option value='az'>A-Z</option>
          <option value='za'>Z-A</option>
          <option value='ascending'>Price - ascending</option>
          <option value='descending'>Price - descending</option>
        </select>
      </div>

      <div className='filter-type'>
        Filter
        <select value={type} onChange={handleFilterProducts}>
          <option value='all'>All</option>
          <option value='motherboard'>Motherboard</option>
          <option value='CPU'>CPU</option>
          <option value='GPU'>GPU</option>
          <option value='PSU'>PSU</option>
        </select>
      </div>
    </div>;
}
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJuYW1lcyI6WyJGaWx0ZXIiLCJwcm9wcyIsImNvdW50IiwidHlwZSIsImhhbmRsZVNvcnRQcm9kdWN0cyIsImhhbmRsZUZpbHRlclByb2R1Y3RzIiwic29ydCJdLCJzb3VyY2VzIjpbIi4uLy4uL3NyYy9jb21wb25lbnRzL0ZpbHRlci50c3giXSwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0IGRlZmF1bHQgZnVuY3Rpb24gRmlsdGVyKHByb3BzOiBhbnkpIHtcclxuICBjb25zdCB7IGNvdW50LCB0eXBlLCBoYW5kbGVTb3J0UHJvZHVjdHMsIGhhbmRsZUZpbHRlclByb2R1Y3RzLCBzb3J0IH0gPSBwcm9wcztcclxuICByZXR1cm4gKFxyXG4gICAgPGRpdj5cclxuICAgICAgPGRpdiBjbGFzc05hbWU9J2ZpbHRlci1yZXN1bHQnPntjb3VudH0gUHJvZHVjdChzKTwvZGl2PlxyXG4gICAgICA8ZGl2IGNsYXNzTmFtZT0nZmlsdGVyLXNvcnQnPlxyXG4gICAgICAgIE9yZGVyeycgJ31cclxuICAgICAgICA8c2VsZWN0IHZhbHVlPXtzb3J0fSBvbkNoYW5nZT17aGFuZGxlU29ydFByb2R1Y3RzfT5cclxuICAgICAgICAgIDxvcHRpb24gdmFsdWU9J2F6Jz5BLVo8L29wdGlvbj5cclxuICAgICAgICAgIDxvcHRpb24gdmFsdWU9J3phJz5aLUE8L29wdGlvbj5cclxuICAgICAgICAgIDxvcHRpb24gdmFsdWU9J2FzY2VuZGluZyc+UHJpY2UgLSBhc2NlbmRpbmc8L29wdGlvbj5cclxuICAgICAgICAgIDxvcHRpb24gdmFsdWU9J2Rlc2NlbmRpbmcnPlByaWNlIC0gZGVzY2VuZGluZzwvb3B0aW9uPlxyXG4gICAgICAgIDwvc2VsZWN0PlxyXG4gICAgICA8L2Rpdj5cclxuXHJcbiAgICAgIDxkaXYgY2xhc3NOYW1lPSdmaWx0ZXItdHlwZSc+XHJcbiAgICAgICAgRmlsdGVyXHJcbiAgICAgICAgPHNlbGVjdCB2YWx1ZT17dHlwZX0gb25DaGFuZ2U9e2hhbmRsZUZpbHRlclByb2R1Y3RzfT5cclxuICAgICAgICAgIDxvcHRpb24gdmFsdWU9J2FsbCc+QWxsPC9vcHRpb24+XHJcbiAgICAgICAgICA8b3B0aW9uIHZhbHVlPSdtb3RoZXJib2FyZCc+TW90aGVyYm9hcmQ8L29wdGlvbj5cclxuICAgICAgICAgIDxvcHRpb24gdmFsdWU9J0NQVSc+Q1BVPC9vcHRpb24+XHJcbiAgICAgICAgICA8b3B0aW9uIHZhbHVlPSdHUFUnPkdQVTwvb3B0aW9uPlxyXG4gICAgICAgICAgPG9wdGlvbiB2YWx1ZT0nUFNVJz5QU1U8L29wdGlvbj5cclxuICAgICAgICA8L3NlbGVjdD5cclxuICAgICAgPC9kaXY+XHJcbiAgICA8L2Rpdj5cclxuICApO1xyXG59XHJcbiJdLCJtYXBwaW5ncyI6Ijs7Ozs7O0FBQWUsU0FBU0EsTUFBTUEsQ0FBQ0MsS0FBVSxFQUFFO0VBQ3pDLElBQVFDLEtBQUssR0FBMkRELEtBQUssQ0FBckVDLEtBQUs7SUFBRUMsSUFBSSxHQUFxREYsS0FBSyxDQUE5REUsSUFBSTtJQUFFQyxrQkFBa0IsR0FBaUNILEtBQUssQ0FBeERHLGtCQUFrQjtJQUFFQyxvQkFBb0IsR0FBV0osS0FBSyxDQUFwQ0ksb0JBQW9CO0lBQUVDLElBQUksR0FBS0wsS0FBSyxDQUFkSyxJQUFJO0VBQ25FLE9BQ0UsQ0FBQyxHQUFHLENBQUM7QUFDVCxNQUFNLENBQUMsR0FBRyxDQUFDLFNBQVMsQ0FBQyxlQUFlLENBQUMsQ0FBQ0osS0FBSyxDQUFDLFdBQVcsRUFBRSxHQUFHLENBQUM7QUFDN0QsTUFBTSxDQUFDLEdBQUcsQ0FBQyxTQUFTLENBQUMsYUFBYSxDQUFDO0FBQ25DLGFBQWEsQ0FBQyxHQUFHLENBQUM7QUFDbEIsUUFBUSxDQUFDLE1BQU0sQ0FBQyxLQUFLLENBQUMsQ0FBQ0ksSUFBSSxDQUFDLENBQUMsUUFBUSxDQUFDLENBQUNGLGtCQUFrQixDQUFDLENBQUM7QUFDM0QsVUFBVSxDQUFDLE1BQU0sQ0FBQyxLQUFLLENBQUMsSUFBSSxDQUFDLEdBQUcsRUFBRSxNQUFNLENBQUM7QUFDekMsVUFBVSxDQUFDLE1BQU0sQ0FBQyxLQUFLLENBQUMsSUFBSSxDQUFDLEdBQUcsRUFBRSxNQUFNLENBQUM7QUFDekMsVUFBVSxDQUFDLE1BQU0sQ0FBQyxLQUFLLENBQUMsV0FBVyxDQUFDLGlCQUFpQixFQUFFLE1BQU0sQ0FBQztBQUM5RCxVQUFVLENBQUMsTUFBTSxDQUFDLEtBQUssQ0FBQyxZQUFZLENBQUMsa0JBQWtCLEVBQUUsTUFBTSxDQUFDO0FBQ2hFLFFBQVEsRUFBRSxNQUFNLENBQUM7QUFDakIsTUFBTSxFQUFFLEdBQUcsQ0FBQztBQUNaO0FBQ0EsTUFBTSxDQUFDLEdBQUcsQ0FBQyxTQUFTLENBQUMsYUFBYSxDQUFDO0FBQ25DO0FBQ0EsUUFBUSxDQUFDLE1BQU0sQ0FBQyxLQUFLLENBQUMsQ0FBQ0QsSUFBSSxDQUFDLENBQUMsUUFBUSxDQUFDLENBQUNFLG9CQUFvQixDQUFDLENBQUM7QUFDN0QsVUFBVSxDQUFDLE1BQU0sQ0FBQyxLQUFLLENBQUMsS0FBSyxDQUFDLEdBQUcsRUFBRSxNQUFNLENBQUM7QUFDMUMsVUFBVSxDQUFDLE1BQU0sQ0FBQyxLQUFLLENBQUMsYUFBYSxDQUFDLFdBQVcsRUFBRSxNQUFNLENBQUM7QUFDMUQsVUFBVSxDQUFDLE1BQU0sQ0FBQyxLQUFLLENBQUMsS0FBSyxDQUFDLEdBQUcsRUFBRSxNQUFNLENBQUM7QUFDMUMsVUFBVSxDQUFDLE1BQU0sQ0FBQyxLQUFLLENBQUMsS0FBSyxDQUFDLEdBQUcsRUFBRSxNQUFNLENBQUM7QUFDMUMsVUFBVSxDQUFDLE1BQU0sQ0FBQyxLQUFLLENBQUMsS0FBSyxDQUFDLEdBQUcsRUFBRSxNQUFNLENBQUM7QUFDMUMsUUFBUSxFQUFFLE1BQU0sQ0FBQztBQUNqQixNQUFNLEVBQUUsR0FBRyxDQUFDO0FBQ1osSUFBSSxFQUFFLEdBQUcsQ0FBQztBQUVWIn0=