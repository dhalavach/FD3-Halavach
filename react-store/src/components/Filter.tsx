import useSearchParamsState from '../hooks/useSearchParamsState';

export default function Filter() {
  const [filterParam, setFilterParam] = useSearchParamsState('filterParam', '');
  const [sortParam, setSortParam] = useSearchParamsState('sortParam', '');

  return (
    <div>
      <div className='filter-sort'>
        Order{' '}
        <select
          value={sortParam}
          onChange={(e) => setSortParam(e.target.value)}
        >
          <option value=''></option>
          <option value='az'>A-Z</option>
          <option value='za'>Z-A</option>
          <option value='ascending'>Price - ascending</option>
          <option value='descending'>Price - descending</option>
        </select>
      </div>

      <div className='filter-type'>
        Filter
        <select
          value={filterParam}
          onChange={(e) => setFilterParam(e.target.value)}
        >
          <option value=''>All</option>
          <option value='motherboard'>Motherboard</option>
          <option value='CPU'>CPU</option>
          <option value='GPU'>GPU</option>
          <option value='PSU'>PSU</option>
        </select>
      </div>
    </div>
  );
}
