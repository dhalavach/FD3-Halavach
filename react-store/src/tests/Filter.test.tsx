import '@testing-library/jest-dom';
import Filter from '../components/Filter';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import store from '../components/Store';

test('component renders without throwing an error', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Filter />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toBeTruthy();
});

test('component matches the snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Filter />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
