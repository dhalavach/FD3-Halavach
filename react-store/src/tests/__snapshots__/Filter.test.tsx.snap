// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches the snapshot 1`] = `
<div>
  <div
    className="filter-sort"
  >
    Order
     
    <select
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      />
      <option
        value="az"
      >
        A-Z
      </option>
      <option
        value="za"
      >
        Z-A
      </option>
      <option
        value="ascending"
      >
        Price - ascending
      </option>
      <option
        value="descending"
      >
        Price - descending
      </option>
    </select>
  </div>
  <div
    className="filter-type"
  >
    Filter
    <select
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
        All
      </option>
      <option
        value="motherboard"
      >
        Motherboard
      </option>
      <option
        value="CPU"
      >
        CPU
      </option>
      <option
        value="GPU"
      >
        GPU
      </option>
      <option
        value="PSU"
      >
        PSU
      </option>
    </select>
  </div>
</div>
`;
